### DISCUSSION TITLE
[Feature] Disable certain external libraries showing in the main photo timeline

### DISCUSSION THREAD

**Discussion by MergenStudios** [immich-app/immich#9380] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It would be convenient to be able to hide certain external libraries from showing up in the main timeline.

I have screenshots from my phone and various screenshots from games saved in external libraries and imported into immich, because they have a time and date attached to them. But while scrolling through my timeline, often I only want to see photos, and not screenshots.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by danieldietzler** [comment]
I think this is a duplicate of #5158 as libraries shouldn't be used for this kind of separation (and in fact that concept may get removed in the future), so albums make much more sense to group those.

**Reply by revilo951** [reply]
Not a duplicate, as I have the same issue as OP - I have an external library that I want to be able to view in immich via the folder browser, but I don't want it cluttering up my main timeline.

I suppose if an external library could be automatically made into an album, and that album excluded from the timeline, that would work.

**Reply by PJGitHub9** [reply]
> Not a duplicate, as I have the same issue as OP - I have an external library that I want to be able to view in immich via the folder browser, but I don't want it cluttering up my main timeline.
> 
> I suppose if an external library could be automatically made into an album, and that album excluded from the timeline, that would work.

I agree with this. With external libraries Id like to choose which libraries are viewable in the main timeline, if not selected, then they can still be viewed in albums or folders, but not in the main timeline (wont show up in memories either)

**Reply by gschintgen** [reply]
This feature would also be very useful for a mixed usecase in combination with e.g. Digikam:

- Use background backup for everyday operation and immediate cross-device availability.
- Periodically, e.g. after a vacation, transfer/manage/tag/edit/curate/etc in Digikam and the re-add the finalized collection in Immich as read-only external library.
- Remove the originals in Immich's upload library.

It's that last step that could be made more convenient by temporarily hiding external libraries: It'd be easier to see which assets are only the unmanaged backups in the default upload library.

I know that may not be the intended all-Immich workflow, but I'd guess that a sizable number of Immich users do try to make it work in combination with external management tools.

A second usecase for me personally is having a separation between my family photos (1-2 decades), my "earlier" life, and family archive pictures (i.e. previous generations). Those are three separate collections, so neither albums nor user accounts would make sense to implement that semantic differentiation.

@danieldietzler : Which concept exactly may disappear?

**Reply by danieldietzler** [reply]
> Which concept exactly may disappear?

The line/difference between external libraries and the default upload library. The plan is to streamline both

**Reply by gschintgen** [reply]
> > Which concept exactly may disappear?
> 
> The line/difference between external libraries and the default upload library. The plan is to streamline both

Thanks for the reply! Will it still be possible to use an external read-only folder for additional pictures? I.e. like today with external libraries (only with some kind of unified implementation)?

**Reply by danieldietzler** [reply]
Yes!

**Comment by MergenStudios** [comment]
Just an update for the people also waiting for this feature:
I recently talked to some of the immich devs at fosdem, and asked them about this exact feature, to which their response (hopefully correctly paraphrased by me) was:

The timeline is the most complex part of the entire interface, and it needs to be really really fast to ensure scrolling through your photos is as seamless as possible. Adding a bunch of filters to the sql query populating the timeline would make it really slow, which is why hiding external libraries or even albums / specific photos from the timeline isnt a thing.
They did add that they are planning to rework the concept of libraries and external libraries as a whole though, and that this might bring the possibility for *some* filtering of photos in the timeline.
